src/Post_Command.php

```diff
@@ -173,6 +180,21 @@ public function create( $args, $assoc_args ) {
 		$array_arguments = array( 'meta_input' );
 		$assoc_args      = \WP_CLI\Utils\parse_shell_arrays( $assoc_args, $array_arguments );
 
+            if( isset( $assoc_args['from-post'] ) ) {
+                  $post     = $this->fetcher->get_check( $assoc_args['from-post'] );
+                  $post_arr = get_object_vars( $post );
+                  $post_id  = $post_arr['ID'];
+                  unset( $post_arr['post_date'] );
+                  unset( $post_arr['post_date_gmt'] );
+                  unset( $post_arr['guid'] );
+
+                  if ( empty( $assoc_args['meta_input'] ) ) {
+                        $assoc_args['meta_input'] = $this->get_metadata( $post_id );
+                  }
```

Comment:
Do we need to persist taxonomy data too?

Reply:
This part is for overwriting existing metadata if a user wants. Categories and tags can be duplicated too. what are your thoughts about it?

Reply:
Yes, I think categories and tags should be duplicated too.